### ISSUE TITLE
Self-hosted bundle analyzer

### ISSUE THREAD

**Issue by fabiospampinato** [evanw/esbuild#2847]
IMO the bundle analyzer should be self-hosted, and not reliant on a website, for a bunch of reasons:

- If I have no internet connection I would be nice to still be able to use it. Alternatively the website could be made to work offline after the first load.
- It'd be easier to automate the generation of bundle visualization. If I have to manually drag & drop a file to get it to work it  gets a little more annoying to do.
- As far as I know the mainstream alternatives for other bundlers ([`webpack-bundle-analyzer`](https://github.com/webpack-contrib/webpack-bundle-analyzer), [`rollup-plugin-visualizer`](https://github.com/btd/rollup-plugin-visualizer)) have the above capabilities, so it would be nice to close the gap.

**Comment by evanw** [comment]
I agree. I think it makes the most sense for this to also be a part of esbuild's dev server eventually.

**Comment by retorquere** [comment]
Where does the source for the visualizer currently live?

**Comment by lydell** [comment]
@retorquere See https://github.com/evanw/esbuild/issues/2926

**Comment by fabiospampinato** [comment]
I've written a little CLI thing that basically implements this for quickly inspecting NPM packages: https://twitter.com/fabiospampinato/status/1680735035899428864

**Comment by acidoxee** [comment]
This would be great to have, for instance to host on GitHub Pages for every PR, or even to have a simple link with the metafile's content in the URL to be able to open the analyzer with the proper data in just one click. I've already started working on a GitHub action to add PR comments to summarize changes (similar to [esbuild-bundle-analyzer](https://github.com/exoego/esbuild-bundle-analyzer)), but nothing beats the official analyzer for a deeper dive!

**Comment by vzakharov-rxnt** [comment]
I second. Would be great to have something that works with Angular 18.

**Comment by vzakharov-rxnt** [comment]
> I've written a little CLI thing that basically implements this for quickly inspecting NPM packages: https://github.com/fabiospampinato/banal

What if I want to inspect my own site? Including node modules if they are referenced? Basically the final bundle.

**Comment by fabiospampinato** [comment]
> What if I want to inspect my own site? Including node modules if they are referenced? Basically the final bundle.

@vzakharov-rxnt `npx banal ./path/to/whereever.js` if you want a bundle analysis for a local thing. If that doesn't work for you (maybe you need custom plugins to bundle it?) please open an issue [here](https://github.com/fabiospampinato/banal/issues).

**Comment by vzakharov-rxnt** [comment]
@fabiospampinato I want to analyze the whole project. Specifying path to main.js file does not work. Tech is Angular.

**Comment by Badisi** [comment]
@vzakharov-rxnt,

1. Make sure to use the `@angular-devkit/build-angular:application` builder so that **esbuild** is used
2. Build your Angular application with stats: `ng build --configuration=production --stats-json`
3. Reference that file when using **banal**: `npx banal --metafile=dist/your-app/stats.json`

**Comment by adumesny** [comment]
FYI https://www.npmjs.com/package/esbuild-visualizer does a decent job visualizing things (not as good as webpack-bundle-analyzer but better than website (quick glance) and most importanly local.
